src/renderer/lib/playlist.js

```diff
@@ -0,0 +1,161 @@
+}
+
+Playlist.prototype.next = function () {
+  if (this.hasNext()) {
+    this._position = mod(this._position + 1, this._tracks.length)
+    return this.getCurrent()
+  }
+}
+
+Playlist.prototype.previous = function () {
+  if (this.hasPrevious()) {
+    this._position = mod(this._position - 1, this._tracks.length)
+    return this.getCurrent()
+  }
+}
+
+Playlist.prototype.shuffleEnabled = function () {
+  return this._shuffled
+}
+
+Playlist.prototype.toggleShuffle = function (value) {
+  this._shuffled = (value === undefined ? !this._shuffled : value)
+  this._shuffled ? this._shuffle() : this._unshuffle()
+}
+
+Playlist.prototype.repeatEnabled = function () {
+  return this._repeat
+}
+
+Playlist.prototype.toggleRepeat = function (value) {
+  this._repeat = (value === undefined ? !this._repeat : value)
+}
+
+Playlist.prototype.jumpToFile = function (infoHash, fileIndex) {
+  this.setPosition(this._order.findIndex((i) => {
+    let track = this._tracks[i]
+    return track.infoHash === infoHash && track.fileIndex === fileIndex
+  }))
+  return this.getCurrent()
+}
+
+Playlist.prototype.getCurrent = function () {
+  var position = this.getPosition()
+
+  return position === undefined ? undefined
+    : this._tracks[this._order[position]]
+}
+
+Playlist.prototype.getPosition = function () {
+  if (this._position >= 0 && this._position < this._tracks.length) {
+    return this._position
+  } else return undefined
+}
+
+Playlist.prototype.setPosition = function (position) {
+  this._position = position
+}
+
+Playlist.prototype.getState = function () {
+  return {
+    hasNext: this.hasNext(),
+    hasPrevious: this.hasPrevious(),
+    shuffle: this.shuffleEnabled(),
+    repeat: this.repeatEnabled()
+  }
+}
+
+// =============================================================================
+// Private methods
+// =============================================================================
+
+Playlist.prototype._shuffle = function () {
+  let order = this._order
+  if (!order.length) return
+
+  // Move the current track to the beggining of the playlist
+  swap(order, 0, this._position)
+  this._position = 0
+
+  // Shuffle the rest of the tracks with Fisher-Yates Shuffle
+  for (let i = order.length - 1; i > 0; --i) {
+    let j = Math.floor(Math.random() * i) + 1
+    swap(order, i, j)
+  }
+}
+
+Playlist.prototype._unshuffle = function () {
+  this._position = this._order[this._position]
+  this._order = range(0, this._order.length)
+}
+
+// =============================================================================
+// Utility fuctions
+// =============================================================================
+
+function extractTracks (torrentSummary) {
+  return torrentSummary.files.map((file, index) => ({ file, index }))
+    .filter((object) => TorrentPlayer.isPlayable(object.file))
+    .sort(function (a, b) {
+      if (a.file.name < b.file.name) return -1
+      if (b.file.name < a.file.name) return 1
```

Comment:
I guess sort by name is okay.
I agree that it should be consistent, maybe it would be easier to sort the files initially and stick with that order everywhere else?